Sources/MongoSwift/BSON/BsonValue.swift

```diff
@@ -185,27 +165,75 @@ public struct Binary: BsonValue, Equatable, Codable {
     public let data: Data
 
     /// The binary subtype for this data.
-    public let subtype: BsonSubtype
+    public let subtype: Subtype
+
+    /// Subtypes for BSON Binary values.
+    public enum Subtype: RawRepresentable, Codable {
+        /// Generic binary subtype
+        case generic,
+        /// A function
+        function,
+        /// Binary (old)
+        binaryDeprecated,
+        /// UUID (old)
+        uuidDeprecated,
+        /// UUID (RFC 4122)
+        uuid,
+        /// MD5
+        md5,
+        /// User defined
+        userDefined(UInt8)
```

Comment:
Will this be a catch-all for anything outside of the range of zero (`generic`) to five (`md5`)? The spec says that all values in the range of 0x80 to 0xFF are user-defined, but the actual `userDefined` constant is a single 0x80 value.

We should also ensure that _any_ unsigned 8-bit value is accepted as a sub-type. To my knowledge, no driver restricts the sub-type byte to 0x00-0x05 and 0x80-0xFF. While 0x06-0x7F are effectively undefined in the spec, there may very well be BSON data out there that uses those values.

Reply:
ah, ok, I misunderstood the spec. so I think we could express this in a couple of different ways.

option 1:
```swift
enum Subtype: UInt8 {
    case generic = 0,
    function,
    binaryDeprecated,
    uuidDeprecated,
    md5,
    userDefined = 0x80
}
```

and then in `Binary` we just store a `UInt8` and allow the user to initialize with either a `Subtype` or some custom `UInt8`. 

option 2:
```swift
enum Subtype: RawRepresentable, Codable {
    case generic = 0,
    function,
    binaryDeprecated,
    uuidDeprecated,
    md5,
    userDefined = 0x80,
    custom(UInt8)
}
```

So basically then we would continue storing a `Subtype` on `Binary`, and allow initialization from either a `Subtype` or a `UInt8`, lumping together all custom values under `.custom` (could be named something else.) 

I don't really have a preference between those two. what do you think?

Reply:
Let's go with option 1 above

Reply:
The first option (define enums for documented constants but allow any unsigned 8-bit integer) is consistent with other drivers I've looked at. I vote for that.